workshop/exercise-6/README.md

```diff
@@ -1,86 +1,110 @@
 
-3. The CLI output should include the API Key value. Create an environment variable containing your key.
+Because Knative is built on top of Kubernetes, you can use kubectl along with configuration files to create a new Service representing your application.
+
+1. In the `fib-knative` project you cloned earlier, you should see a file called `fib-service.yaml`. Look at the contents of the file:
 
     ```
-    export MYAPIKEY=<your_api_key_value>
+    cat fib-service.yaml
     ```
 
-### Provide Container Registry Credentials to Cluster
-This lab will need credentials for authenticating to your private container registry. First, we'll need to create a `Secret` to store the credentials for this registry. This secret will be used for the knative-serving component to pull down an image from the container registry.
+    File Contents:
+    ```
+    apiVersion: serving.knative.dev/v1
+    kind: Service
+    metadata:
+      name: fib-knative
+      namespace: default
+    spec:
+      template:
+        metadata:
+          name: fib-knative-one
+        spec:
+          containers:
+            - image: docker.io/ibmcom/fib-knative
+    ```
 
-A `Secret` is a Kubernetes object containing sensitive data such as a password, a token, or a key. You can also read more about [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/).
+    The `fib-service.yaml` file describes a Service. Notice that it includes the name (fib-knative), the namespace (default), and a reference to the container image on dockerhub (docker.io/ibmcom/fib-knative). Take a look at the `template` section of the Service spec. Any time this is updated, a new revision is created. In this case, our first revision will be named fib-knative-one.
 
-1. Let's create a secret, which will be a `docker-registry` type secret. This type of secret is used to authenticate with a container registry to pull down a private image. We can create this via the commandline. For username, simply use the string `iamapikey`. For `<api_key_value>`, use the api key you made note of earlier.
+2. Let's deploy this app into our cluster. Apply the `fib-service.yaml` file.
 
     ```
-    kubectl --namespace default create secret docker-registry ibm-cr-secret  --docker-server=us.icr.io --docker-username=iamapikey --docker-password=$MYAPIKEY
+    kubectl apply --filename fib-service.yaml
     ```
 
-2. We will also need a secret for the build process to have credentials to push the built image to the container registry. To create this secret, we'll first need to base64 encode our username and password for IBM Container Registry. For username, you will use the string `iamapikey`. The base64 encoding of `iamapikey` should be: `aWFtYXBpa2V5` - which is already in the yaml file.  Let's base64 encode our apikey, and copy it.
+3. Watch the pods initializing as our application gets deployed and starts up:
```

Comment:
I think I'd prefer to re-org this exercise as: Advanced debugging
And then just show how they can use kubectl to look at the pods, etc... but not use kubectl to deploy the app. People might interpret it as "in order to see the pods I need to leave `kn` and use `kubectl`"

Reply:
hrmm that's fine. A little hesitant about not deploying using kubectl since when they go off to knative land that's mostly what they'll see in docs, etc. That said, advanced users will see applying .yaml files in the advanced exercises so this could be okay.